vip-plugins/vip-plugins.php

```diff
@@ -0,0 +1,222 @@
+<?php 
+
+/**
+ * Retrieve featured plugins from the vip.wordpress.com API
+ *
+ * @return array an array of plugins
+ */
+function wpcom_vip_fetch_vip_featured_plugins() {
+	$plugins = wp_cache_get( 'wpcom_vip_featured_plugins' );
+
+	if ( false === $plugins ) {
+		$plugins = array();
+		$url_for_featured_plugins = 'https://vip.wordpress.com/wp-json/vip/v1/plugins?type=technology';
+		$response = vip_safe_wp_remote_get( $url_for_featured_plugins, false, 3, 5 );
+
+		if ( ! $response || is_wp_error( $response ) ) {
+			trigger_error( 'The API on vip.wordpress.com is not responding (' . esc_url( $url_for_featured_plugins ) . ')' );
+			return false;
+		}
+
+		$plugins = json_decode( $response['body'] );
+
+		if ( empty( $plugins ) ) {
+			return false;
+		}
+
+		wp_cache_set( 'wpcom_vip_featured_plugins', $plugins, '', HOUR_IN_SECONDS * 4 );
+	}
+
+	return $plugins;
+}
+
+/**
+ * Render the featured partner plugins to the plugins screenReaderText
+ * Uses the notice hooks as that is all we have on these pages
+ *
+ * @return null
+ */
+function wpcom_vip_render_vip_featured_plugins() {
+	$screen = get_current_screen();
+
+	if ( ! ( 'plugins' === $screen->id || 'plugins-network' === $screen->id ) ) {
+		return;
+	}
+
+	$plugins = wpcom_vip_fetch_vip_featured_plugins();
+
+	if ( ! $plugins ) {
+		return;
+	}
+
+	?>
+	<div class="featured-plugins notice">
+		<h3><?php _e( 'VIP Featured Plugins', 'vip-dashboard' ); ?></h3>
+		<?php
+		foreach ( $plugins as $key => $plugin ) {
+			?>
+			<div class="plugin">
+				<a class="fp-content" href="<?php echo esc_attr( $plugin->meta->plugin_url ); ?>" target="_blank">
+					<img src="<?php echo esc_attr( $plugin->meta->listing_logo ); ?>" alt="<?php echo esc_attr( $plugin->post_title ); ?>" />
+					<h4><?php echo esc_html( $plugin->post_title ); ?></h4>
+					<p><?php echo esc_html( $plugin->meta->listing_description ); ?></p>
+				</a>
+				<a class="fp-overlay" href="<?php echo esc_attr( $plugin->meta->plugin_url ); ?>" target="_blank">
+					<div class="fp-overlay-inner">
+						<div class="fp-overlay-cell">
+							<span>	
+								<?php _e( 'More Information', 'vip-dashboard' ); ?>
+							</span>
+						</div>
+					</div>
+				</a>
+			</div>
+			<?php
+		}
+		?>
+	</div>
+	<?php
+}
+add_action( 'admin_notices', 'wpcom_vip_render_vip_featured_plugins', 99 );
+add_action( 'network_admin_notices', 'wpcom_vip_render_vip_featured_plugins', 99 );
```

Comment:
Minor: would be good to note why we're using `admin_notices` for this and why priority 99.

Reply:
Added to the function notes

Reply:
FWIW, I actually find it's best to have details like this as close to the source as possible. If I'm looking through the code and trying to figure out why priority 99 is being used, my instinct is not to go the callback's docblock to get those details.

Additionally, if we change the priority or the action down the road, it's easier to forget to update the callback docblock since it's not right there, and we end up with outdated inline docs.